### PR TITLE
Add S3 ingress

### LINKED ISSUE #23: Use Ingress as S3 Read Proxy
- [x] I have searched issues for similar feature requests

## What is this feature about (1 sentence)

This feature is about allowing the usage of an `Ingress` resource, combined with a `Service` type `ExternalName`, to expose read access to S3 Bucket through a custom URL (see more about `S3_READ_URL_PREFIX` [here](https://github.com/sorry-cypress/sorry-cypress/pull/53), accessed by user's browser.

## Why is it needed? What is the value? For whom do we build it?

Assuming you want to define a custom URL (based on [`S3_READ_URL_PREFIX`](https://github.com/sorry-cypress/sorry-cypress/pull/53)), allowing read access to S3 Bucket through a private endpoint (VPC network only, for example), you should be able to define it using native K8s resources, as `Ingress` and `Service` to get it done. The approach of using `Service` type `ExternalName` is described in detail on this [Google Cloud Blog post](https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-mapping-external-services). It is a good way to use a [feature provided by sorry-cypress](https://github.com/sorry-cypress/sorry-cypress/pull/53/commits/7b1fa8f2e0d3d4461630fb4e643a0f6694819c3d) without requiring new external infrastructure components.

## High-level feature overview

sorry-cypress-static ingress

Raw manifests example (using nginx as ingress-controller):

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    nginx.ingress.kubernetes.io/upstream-vhost: "<your_bucket_name>.s3-website-<aws_region>.amazonaws.com"
    nginx.ingress.kubernetes.io/enable-cors: "true"
    nginx.ingress.kubernetes.io/cors-allow-origin: ""POST, GET, PUT, DELETE, HEAD"
    kubernetes.io/ingress.class: nginx-aws
  name: sorry-cypress-static
  namespace: sorry-cypress
spec:
  rules:
  - host: <your_bucket_name>.s3-website-<aws_region>.amazonaws.com
    http:
      paths:
      - backend:
          serviceName: sorry-cypress-static
          servicePort: http
        path: /
```
```
apiVersion: v1
kind: Service
metadata:
  name: sorry-cypress-static
  namespace: sorry-cypress
spec:
  externalName: <your_bucket_name>.s3-website-<aws_region>.amazonaws.com
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: 80
  type: ExternalName
```

## Other

*NOTE*: If you want to restrict access to your S3 bucket only from your AWS VPC Network, it is recommended creating a Bucket Policy with `aws:SourceVpc` or `aws:SourceVpce` condition keys. More info on [Amazon S3 Condition Keys](https://docs.aws.amazon.com/AmazonS3/latest/dev/amazon-s3-policy-keys.html). To use that is required the creation of a [S3 VPC Endpoint](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-endpoints-s3.html) inside your VPC.

### PR DESCRIPTION
**Pull request by fsmaia**
Closes #23

### CODE REVIEW

File: charts/sorry-cypress/templates/ingress-s3.tpl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  labels:
+    {{- include "sorry-cypress-helm.labels" . | nindent 4 }}
+    {{- with .Values.s3.ingress.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end}}
+  annotations:
+    nginx.ingress.kubernetes.io/upstream-vhost: {{ include "s3Host" . }}
+    nginx.ingress.kubernetes.io/enable-cors: "true"
+    nginx.ingress.kubernetes.io/cors-allow-origin: "*"
+    {{- with .Values.s3.ingress.annotations }}
```
Comment (jhonatanmorais) [inline charts/sorry-cypress/templates/ingress-s3.tpl:23]: What about add this annotation, `nginx.ingress.kubernetes.io/cors-allow-methods: "POST, GET, PUT, DELETE, HEAD"`, according to [documentation](https://sorry-cypress.dev/installation/aws#manually-setup-s3-bucket-for-screenshots-and-videos)?

**Comment by fsmaia** [comment]
Dunno if we should about Nginx CORS attributes:
1. hardcode it in the template (current solution)
2. move them to values, as default
3. remove them from the code and add notes on README
4. add `s3.ingress.type` value and implement these attributes for several controllers

@jhonatanmorais, @tico24, @agoldis, what are your thoughts here?

**Comment by jhonatanmorais** [comment]
> Dunno if we should about Nginx CORS attributes:
> 
> 1. hardcode it in the template (current solution)
> 2. move them to values, as default
> 3. remove them from the code and add notes on README
> 4. add `s3.ingress.type` value and implement these attributes for several controllers
> 
> @jhonatanmorais, @tico24, @agoldis, what are your thoughts here?

IMHO, number 2 or 4 are the best options. I don't have a preference between them.

**Comment by agoldis** [comment]
@fsmaia I don't have a string opinion, the mindset is that helm users are quire powerful and should be able to figure out the CORS settings. Having the topic documented is important though, regardless ;)

**Review by agoldis — APPROVED** [review]

**Comment by fsmaia** [comment]
> > Dunno if we should about Nginx CORS attributes:
> > 
> > 1. hardcode it in the template (current solution)
> > 2. move them to values, as default
> > 3. remove them from the code and add notes on README
> > 4. add `s3.ingress.type` value and implement these attributes for several controllers
> > 
> > @jhonatanmorais, @tico24, @agoldis, what are your thoughts here?
> 
> IMHO, number 2 or 4 are the best options. I don't have a preference between them.

Hmm, I'll go with the number 2 then. :)
It would be too complex to implement annotations of several different controllers.

**Pull request merged**